Assets/Scripts/PastaParticleControler.cs

```diff
@@ -1,33 +1,41 @@
-﻿using System.Collections;
+﻿using System;
+using System.Collections;
 using System.Collections.Generic;
 using UnityEngine;
 
 public class PastaParticleControler : MonoBehaviour
 {
     [SerializeField] private float speed = 2;
     [SerializeField] private int currentTargetPointer = 1;
+    [SerializeField] private int flawThreshold = 10; //ilość uszkodzonych w procentach
+    [SerializeField] private bool isBroken = false;
 
     public bool movementToggle = true;
     public List<GameObject> AssemblyLineMachinesRoute = new List<GameObject>();
 
     void Start()
     {
         currentTargetPointer = 0;
+        System.Random random = new System.Random();
```

Comment:
Ja to bym se przerzucił deklaracje Random do parametrów w ciele klasy a w Start() tylko zainicjalizował

Reply:
A tak to wszystko gicior